### PR TITLE
[docs] Document database steps

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: docs/modules/plugins/pages/plugin-db.adoc [hunk 1]
```diff
@@ -354,3 +354,135 @@ specified database using SQL query.
...
+
+.Compare name and country columns in result set againser defined data set
+[source,gherkin]
+----
+When I execute SQL query `SELECT * FROM launch_rockets` against `rockets` and save result to scenario variable `all-launches`
+Then `${all-launches}` matching rows using `name,country` from `launches` is equal to data from:
+|name    |country|
+|Falcon 9|USA    |
+|Proton  |RUSSIA |
+----
```
Comment (ikalinin1) [inline docs/modules/plugins/pages/plugin-db.adoc:488]:
```suggestion

== Steps

=== Execute SQL query and save its result

Executes SQL query against specified database and saves the query result to a variable.

[source,gherkin]
----
When I execute SQL query `$sqlQuery` against `$dbKey` and save result to $scopes variable `$variableName`
----

* `$sqlQuery` - https://www.techtarget.com/searchdatamanagement/definition/SQL[The SQL query to execute].
* `$dbKey` - <<_database_connection, The database connection key>>.
* `$scopes` - xref:commons:variables.adoc#_scopes[The comma-separated set of the variables scopes].
* `$variableName` - The variable name to store the SQL query result.

.The launch_rockets table available by 'launches' db connection
[source]
----
|id|name    |payload_cost|country|
|1 |Falcon 9|$2,720      |USA    |
|2 |Proton  |$4,320      |RUSSIA |
----

.Query USA rocket launches and validate result set
[source,gherkin]
----
When I execute SQL query `SELECT * FROM launch_rockets WHERE country='USA'` against `launches` and save result to scenario variable `usa-launches`
Then `usa-launches` is equal to `[{
    name=Falcon 9,
    payload_cost=$2,720,
    country=USA
}]`
----

Individual rows are available by index starting from zero as per the following example:

.Query USA rocket launches and check name in first row
[source,gherkin]
----
When I execute SQL query `SELECT * FROM launch_rockets WHERE country='USA'` against `launches` and save result to scenario variable `usa-launches`
Then `usa-launches[0].name` is equal to `Falcon 9`
----

=== Execute update query

Executes SQL query that changes the target table e.g. https://www.w3schools.com/sql/sql_update.asp[UPDATE], https://www.w3schools.com/sql/sql_insert.asp[INSERT], https://www.w3schools.com/sql/sql_delete.asp[DELETE] SQL queries. In the case of https://www.w3schools.com/sql/sql_select.asp[SELECT] SQL query an exception will be thrown.

[source,gherkin]
----
When I execute SQL query `$sqlQuery` against `$dbKey`
----

* `$sqlQuery` - https://www.techtarget.com/searchdatamanagement/definition/SQL[The SQL query to execute].
* `$dbKey` - <<_database_connection, The database connection key>>.

.The launch_rockets table available by 'launches' db connection
[source]
----
|id|name    |payload_cost|country|
|1 |Falcon 9|$2,720      |USA    |
|2 |Proton  |$4,320      |RUSSIA |
----

.Update payload_cost for 'Falcon 9' launches
[source,gherkin]
----
When I execute SQL query `UPDATE launch_rockets SET payload_cost='unknown' WHERE name='Falcon 9';` against `launches`
----

=== Compare two result sets

Compares result set queried from the baseline database with the result set queries from the checkpoint database by using user defined keys. Please see <<_data_comparison>> for more fine-grained comparison configuration. The order of columns is ignored.

[source,gherkin]
----
Then data from `$leftSqlQuery` executed against `$leftDbKey` $comparisonRule data from `$rightSqlQuery` executed against `$rightDbKey` matching rows using keys:$keys
----

* `$leftSqlQuery` - https://www.techtarget.com/searchdatamanagement/definition/SQL[The SQL query to execute] to retrieve baseline result set.
* `$leftDbKey` - <<_database_connection, The database connection key>> for baseline result set.
* `$comparisonRule` - xref:parameters:comparison-rule.adoc[The comparison rule].
* `$rightSqlQuery` - https://www.techtarget.com/searchdatamanagement/definition/SQL[The SQL query to execute] to retrieve checkpoint result set.
* `$rightDbKey` - <<_database_connection, The database connection key>> for checkpoint result set.
* `$keys` - The database columns to compare.

.The launch_rockets table available by 'launches' and 'launches-replica' db connections
[source]
----
|id|name    |payload_cost|country|
|1 |Falcon 9|$2,720      |USA    |
|2 |Proton  |$4,320      |RUSSIA |
----

.Compare name and country columns in launch_rockets table contained in two databases
[source,gherkin]
----
Then data from `SELECT * FROM launch_rockets` executed against `launches` is equal to data from `SELECT * FROM launch_rockets` executed against `launches-replica` matching rows using keys:name,country
----

=== Compare result set with table

Compares result set queried from the baseline database with the user-defined data set using specified keys. Please see <<_data_comparison>> for more fine-grained comparison configuration. The order of columns is ignored.

[source,gherkin]
----
Then `$leftData` matching rows using `$keys` from `$leftDbKey` $comparisonRule data from:$rightTable
----

* `$leftData` - The baseline result set.
* `$keys` - The database columns to compare.
* `$comparisonRule` - xref:parameters:comparison-rule.adoc[The comparison rule].
* `$rightTable` - The checkpoint data set.

.The launch_rockets table available by 'launches' db connection
[source]
----
|id|name    |payload_cost|country|
|1 |Falcon 9|$2,720      |USA    |
|2 |Proton  |$4,320      |RUSSIA |
----

.Compare name and country columns in result set against defined data set
[source,gherkin]
----
When I execute SQL query `SELECT * FROM launch_rockets` against `rockets` and save result to scenario variable `all-launches`
Then `${all-launches}` matching rows using `name,country` from `launches` is equal to data from:
|name    |country|
|Falcon 9|USA    |
|Proton  |RUSSIA |
----
```

File: docs/modules/plugins/pages/plugin-db.adoc [hunk 2]
```diff
@@ -354,3 +354,135 @@ specified database using SQL query.
...
+When I execute SQL query `$sqlQuery` against `$dbKey` and save result to $scopes variable `$variableName`
+----
+
+* `$sqlQuery` - https://www.techtarget.com/searchdatamanagement/definition/SQL[The SQL query to execute].
+* `$dbKey` - <<_database_connection, The database connection key>>.
+* `$scopes` - xref:commons:variables.adoc#_scopes[The comma-separated set of the variables scopes].
+* `$variableName` - The variable name to store the SQL query result.
+
+.The launch_rockets table available by 'launches' db connection
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-db.adoc:374]:
```suggestion
.Imagine `launch_rockets` table available at 'launches' db connection
```

File: docs/modules/plugins/pages/plugin-db.adoc [hunk 3]
```diff
@@ -354,3 +354,135 @@ specified database using SQL query.
...
+|id|name        |payload_cost|country|
+|1 |Falcon 9    |$2,720      |USA    |
+|2 |Zhongxing-19|$4,320      |CHINA  |
+----
+
+.Query USA rocket launches and validate result set
+[source,gherkin]
+----
+When I execute SQL query `SELECT * FROM launch_rockets WHERE country='USA'` against `launches` and save result to scenario variable `usa-launches`
+Then `usa-launches` is equal to `[{
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-db.adoc:386]: Let's use ExamplesTable here

File: docs/modules/plugins/pages/plugin-db.adoc [hunk 4]
```diff
@@ -354,3 +354,135 @@ specified database using SQL query.
...
+.Query USA rocket launches and check name in first row
+[source,gherkin]
+----
+When I execute SQL query `SELECT * FROM launch_rockets WHERE country='USA'` against `launches` and save result to scenario variable `usa-launches`
+Then `usa-launches[0].name` is equal to `Falcon 9`
+----
+
+=== Execute update query
+
+Executes SQL query that changes the target table e.g. https://www.w3schools.com/sql/sql_update.asp[UPDATE], https://www.w3schools.com/sql/sql_insert.asp[INSERT], https://www.w3schools.com/sql/sql_delete.asp[DELETE] SQL queries. In case of https://www.w3schools.com/sql/sql_select.asp[SELECT] SQL query an exception will be thrown.
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-db.adoc:404]: Use list to enumerate commands

File: docs/modules/plugins/pages/plugin-db.adoc [hunk 5]
```diff
@@ -354,3 +354,135 @@ specified database using SQL query.
...
+
+[source,gherkin]
+----
+When I execute SQL query `$sqlQuery` against `$dbKey`
+----
+
+* `$sqlQuery` - https://www.techtarget.com/searchdatamanagement/definition/SQL[The SQL query to execute].
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-db.adoc:411]:
```suggestion
* `$sqlQuery` - The https://www.techtarget.com/searchdatamanagement/definition/SQL[SQL query to execute].
```

File: docs/modules/plugins/pages/plugin-db.adoc [hunk 6]
```diff
@@ -354,3 +354,135 @@ specified database using SQL query.
...
+* `$dbKey` - <<_database_connection, The database connection key>>.
+
+.The launch_rockets table available by 'launches' db connection
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-db.adoc:414]: See comment above

File: docs/modules/plugins/pages/plugin-db.adoc [hunk 7]
```diff
@@ -354,3 +354,135 @@ specified database using SQL query.
...
+    country=USA
+}]`
+----
+
+The individual rows are available by index starting from zero as per the following example:
+
+.Query USA rocket launches and check name in first row
+[source,gherkin]
+----
+When I execute SQL query `SELECT * FROM launch_rockets WHERE country='USA'` against `launches` and save result to scenario variable `usa-launches`
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-db.adoc:398]: Please format SQL query

File: docs/modules/plugins/pages/plugin-db.adoc [hunk 8]
```diff
@@ -354,3 +354,135 @@ specified database using SQL query.
...
+
+.Update payload_cost for 'Falcon 9' launche
+[source,gherkin]
+----
+When I execute SQL query `UPDATE launch_rockets SET payload_cost='unknown' WHERE name='Falcon 9';` against `launches`
+----
+
+=== Compare two result sets
+
+Compares result set queried from the baseline database with the result set queries from the checkpoint database by using user defined keys. Please see <<_data_comparison>> for more fine-grained comparison configuration. The order of columns is ignored.
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-db.adoc:430]:
```suggestion
Compares the result set queried from the baseline database with the result set queried from the checkpoint database matching by user-defined keys. Please see <<_data_comparison>> for more fine-grained comparison configuration. The order of columns is ignored.
```

File: docs/modules/plugins/pages/plugin-db.adoc (hunk 2 above)
Comment (valfirst) [reply docs/modules/plugins/pages/plugin-db.adoc:374]: yep

File: docs/modules/plugins/pages/plugin-db.adoc (hunk 7 above)
Comment (uarlouski) [reply docs/modules/plugins/pages/plugin-db.adoc]: Formatted for cases where we have where statement, just SELECT * FROM launch_rockets were not formatted

**Review by valfirst — APPROVED** [review]

**Pull request merged**